### PR TITLE
Remove getopt (os specific)

### PR DESCRIPTION
**Pull request by krocard**
Test platform was only accepting one option. Porting getopt for windows seemed overcomplicated compared to searching for the option in argv.

Remove getopt and refactor some utility function to ease error formating.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/195%23issuecomment-139264589%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/195%23discussion_r39170589%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/195%23discussion_r39170841%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/195%23issuecomment-139275417%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/195%23discussion_r39171747%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/195%23discussion_r39194701%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/195%23discussion_r39194728%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/195%23discussion_r39196050%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/195%23issuecomment-139339714%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/195%23issuecomment-139264589%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%20%40dawagner%20%40OznOg%20please%20review%22%2C%20%22created_at%22%3A%20%222015-09-10T14%3A33%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%3Asheep%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-10T15%3A11%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40dawagner%20fixed%20your%20remarks.%22%2C%20%22created_at%22%3A%20%222015-09-10T18%3A43%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%206f70c739ef2bf714d7164415b47db84659106e21%20test/test-platform/main.cpp%2079%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/195%23discussion_r39170841%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22coding%20style%22%2C%20%22created_at%22%3A%20%222015-09-10T15%3A06%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-10T18%3A22%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/test-platform/main.cpp%3AL29-113%22%7D%2C%20%22Pull%202dbfce75e11abce832a027f2664fb7116bf26cef%20test/test-platform/main.cpp%2049%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/195%23discussion_r39171747%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22what%20about%20--help%20%3F%22%2C%20%22created_at%22%3A%20%222015-09-10T15%3A14%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-10T18%3A31%3A29Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/test-platform/main.cpp%3AL29-113%22%7D%2C%20%22Pull%206f70c739ef2bf714d7164415b47db84659106e21%20test/test-platform/main.cpp%2033%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/195%23discussion_r39170589%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22matbe%20pass%20ref%22%2C%20%22created_at%22%3A%20%222015-09-10T15%3A04%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-10T18%3A21%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/test-platform/main.cpp%3AL29-113%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/195#issuecomment-139264589'>General Comment</a></b>
- <a href='https://github.com/krocard'></a> :+1: @dawagner @OznOg please review
- <a href='https://github.com/OznOg'></a> :+1: :sheep:
- <a href='https://github.com/krocard'></a> @dawagner fixed your remarks.
- [x] <a href='#crh-comment-Pull 6f70c739ef2bf714d7164415b47db84659106e21 test/test-platform/main.cpp 33'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/195#discussion_r39170589'>File: test/test-platform/main.cpp:L29-113</a></b>
- <a href='https://github.com/OznOg'></a> matbe pass ref
- [x] <a href='#crh-comment-Pull 6f70c739ef2bf714d7164415b47db84659106e21 test/test-platform/main.cpp 79'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/195#discussion_r39170841'>File: test/test-platform/main.cpp:L29-113</a></b>
- <a href='https://github.com/OznOg'></a> coding style
- [x] <a href='#crh-comment-Pull 2dbfce75e11abce832a027f2664fb7116bf26cef test/test-platform/main.cpp 49'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/195#discussion_r39171747'>File: test/test-platform/main.cpp:L29-113</a></b>
- <a href='https://github.com/dawagner'></a> what about --help ?

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/195?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/195?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/195'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/195'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by krocard** [comment]
:+1: @dawagner @OznOg please review

File: test/test-platform/main.cpp [hunk 1]
```diff
@@ -29,70 +29,84 @@
...
 
 static void showUsage()
 {
-    cerr << "test-platform [-h] <file path> [port number, default "
+    cerr << "test-platform [-h|--help] <file path> [port number, default "
          << iDefaultPortNumber << "]" << endl;
 }
 
-static void showInvalidUsage()
+static void showInvalidUsage(string error)
```
Comment (OznOg) [inline test/test-platform/main.cpp]: matbe pass ref

File: test/test-platform/main.cpp [hunk 2]
```diff
@@ -29,70 +29,84 @@
...
-        default:
-            showInvalidUsage();
-            return -1;
-        }
+    using Options = std::list<string>;
+    // argv[0] is the program name, not an option
+    Options options(argv + 1, argv + argc);
+
+    // Handle help option
+    auto helpOpts = {"-h", "--help"};
```
Comment (OznOg) [inline test/test-platform/main.cpp]: coding style

**Comment by OznOg** [comment]
:+1: :sheep:

File: test/test-platform/main.cpp [hunk 3]
```diff
@@ -29,70 +29,84 @@
...
 static void showHelp()
 {
     showUsage();
-    cerr << "<file path> must be a valid .xml file, oftenly ParameterFrameworkConfiguration.xml" << endl;
-    cerr << "Arguments:" << endl
-        << "    -h  display this help and exit" << endl;
+    cerr << "<file path> must be a valid Paramter top level config file, "
+         << "often named ParameterFrameworkConfiguration.xml.\n"
+         << "Arguments:" << endl
+         << "    -h  display this help and exit" << endl;
```
Comment (dawagner) [inline test/test-platform/main.cpp]: what about --help ?

File: test/test-platform/main.cpp (hunk 1 above)
Comment (krocard) [inline test/test-platform/main.cpp]: :+1:

File: test/test-platform/main.cpp (hunk 2 above)
Comment (krocard) [inline test/test-platform/main.cpp]: :+1:

File: test/test-platform/main.cpp (hunk 3 above)
Comment (krocard) [inline test/test-platform/main.cpp]: :+1:

**Comment by krocard** [comment]
@dawagner fixed your remarks.

**Pull request merged**